x-pack/plugin/core/src/main/java/org/elasticsearch/license/StartTrialClusterTask.java

```diff
@@ -54,20 +56,21 @@ public class StartTrialClusterTask implements ClusterStateTaskListener {
         this.request = request;
         this.listener = listener;
         this.clock = clock;
+        this.featureService = featureService;
     }
 
     private LicensesMetadata execute(
         LicensesMetadata currentLicensesMetadata,
-        DiscoveryNodes discoveryNodes,
+        ClusterState state,
         ClusterStateTaskExecutor.TaskContext<StartTrialClusterTask> taskContext
     ) {
         assert taskContext.getTask() == this;
-        if (discoveryNodes.getMaxNodeVersion().after(discoveryNodes.getSmallestNonClientNodeVersion())) {
+        if (featureService.clusterHasFeature(state, License.INDEPENDENT_TRIAL_VERSION_FEATURE) == false) {
             throw new IllegalStateException(
                 "Please ensure all nodes are on the same version before starting your trial, the highest node version in this cluster is ["
-                    + discoveryNodes.getMaxNodeVersion()
+                    + state.nodes().getMaxNodeVersion()
```

Comment:
I wonder if this error message could be simplified to not note the versions? As it is we are still relying on Version here.

Reply:
We are, but since this is in the Start Trial API, this error message is specific to on-prem deployments, where AFAIK we'll be using traditional version numbers for releases for the foreseeable future and communication with customers in that situation seems like an acceptable use for the traditional version. I can remove it though; not like it's difficult to figure out the current node version.

Reply:
Even if it's on-prem, having uses of Version means we can't remove Version as a thing that can be compared. As long as Version exists as it is today, we run the risk that comparisons leak into serverless which can cause upgrades to fail. If we just need the string, you can get that for the current node from Build.current().

@thecoop Is there a way to get the string version from other nodes through DiscoveryNodes?

Reply:
Hmm, I don't believe so. It is available in the main response of each node, but currently not through DiscoveryNode